jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/solution/route/activity/AbstractActivityNEW.java

```diff
@@ -0,0 +1,128 @@
+package com.graphhopper.jsprit.core.problem.solution.route.activity;
+
+
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import com.graphhopper.jsprit.core.problem.Capacity;
+import com.graphhopper.jsprit.core.problem.Location;
+
+public abstract class AbstractActivityNEW implements TourActivity {
+
+    private int index;
+    protected Capacity capacity;
+    protected double arrTime;
+    protected double endTime;
+    protected double theoreticalEarliest = 0;
+    protected double theoreticalLatest = Double.MAX_VALUE;
+    protected String name;
+    protected Location location;
+
+
+    public AbstractActivityNEW(String name, Location location, Capacity capacity) {
+        super();
```

Comment:
Good idea. The big question: do we need to handle assigned and unassigned activities together anywhere (such as in collections) or not? If yes, I would suggest to use only additional fields in activites for the "runtime" data (as we use the arrTime and endTime now). If no, we could think of a wrapper for these runtime data.